contracts/base/LenderInfo.sol

```diff
@@ -0,0 +1,216 @@
+/*
+    Copyright 2020 Fabrx Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+pragma solidity 0.5.17;
+
+// Libraries
+import "../util/AddressLib.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+// Commons
+import "../util/ZeroCollateralCommon.sol";
+
+// Interfaces
+import "../interfaces/DAIPoolInterface.sol";
+import "../interfaces/LenderInfoInterface.sol";
+import "../interfaces/ZDAIInterface.sol";
+
+contract LenderInfo is LenderInfoInterface {
+    using AddressLib for address;
+    using SafeMath for uint256;
+
+    /* State Variables */
+
+    // last block number of accrued interest
+    uint256 public blockAccruedInterest;
+
+    DAIPoolInterface public daiPool;
+
+    ZDAIInterface public zdai;
+
+    // array of all lending accounts
+    mapping (address => ZeroCollateralCommon.LendAccount) public lenderAccounts;
+
+    /** Modifiers */
+
+    modifier isZDai(address anAddress) {
+        require(areAddressesEqual(address(zdai), anAddress), "Address has no permissions.");
+        _;
+    }
+
+    modifier isDaiPool(address anAddress) {
+        require(areAddressesEqual(address(daiPool), anAddress), "Address has no permissions.");
+        _;
+    }
+
+    modifier isValid(address anAddress) {
+        anAddress.requireNotEmpty("Address is required.");
+        _;
+    }
+
+    /* Constructor */
+
+    constructor(
+        address zdaiAddress,
+        address daiPoolAddress
+    )
+    public
+    {
+        require(zdaiAddress != address(0x0), "ZDai address is required.");
+        require(daiPoolAddress != address(0x0), "Dai pool address is required.");
+        zdai = ZDAIInterface(zdaiAddress);
+        daiPool = DAIPoolInterface(daiPoolAddress);
+        blockAccruedInterest = block.number;
+    }
+
+    /** External Functions */
+
+    function zDaiTransfer(address sender, address recipient, uint256)
```

Comment:
If the `uint256` isn't needed in the contract then feel free to remove it from both the contract and the interface too. I just added it here in case it was needed.

Reply:
Perfect. Let me confirm it.

Reply:
Any update on this?

Reply:
Any update on this?

Reply:
Yes, as the team is analyzing how the interest will be calculated, I think the amount will be required. But the new calculation will be developed in a new PR once It is defined.
I have created a task so we can track it.